akka-stream/src/main/scala/akka/stream/impl/fusing/ActorGraphInterpreter.scala

```diff
@@ -447,15 +449,34 @@ import scala.util.control.NonFatal
   private var self: ActorRef = _
   lazy val log = Logging(mat.system.eventStream, self)
 
-  final case class AsyncInput(shell: GraphInterpreterShell, logic: GraphStageLogic, evt: Any, handler: (Any) ⇒ Unit) extends BoundaryEvent {
+  /**
+   *
+   * @param promise Will be completed upon
+   */
```

Comment:
What a cliff hanger.

Reply:
It's not completed yet, that is the only thing that is for sure.